specification/cognitiveservices/ContentSafety/models.tsp

```diff
@@ -236,3 +236,175 @@ model JailbreakAnalysisResult {
   @doc("Analysis result for jailbreak.")
   detected: boolean;
 }
+
+#suppress "@azure-tools/typespec-azure-core/documentation-required" "MUST fix in next update"
+@added(ContentSafety.Versions.v2024_02_15_Preview)
+@doc("Groundedness Domain.")
+union GroundednessDomain {
+  string,
+  Generic: "Generic",
+  Medical: "Medical",
+}
+
+#suppress "@azure-tools/typespec-azure-core/documentation-required" "MUST fix in next update"
+@added(ContentSafety.Versions.v2024_02_15_Preview)
+@doc("Groundedness Task type.")
+union GroundednessTask {
+  string,
+  Summarization: "Summarization",
+  QnA: "QnA",
+}
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Property types must use PascalCase"
+@added(ContentSafety.Versions.v2024_02_15_Preview)
+@doc("The request of QnA options.")
+model QnAOptions {
+  @doc("The user's question input in a QnA scenario.")
+  @maxLength(7500)
+  query: string;
+}
+
+@added(ContentSafety.Versions.v2024_02_15_Preview)
+@doc("LLM resource type.")
+union LLMResourceType {
+  string,
+
+  @doc("Azure OpenAI resource type.")
+  AzureOpenAI: "AzureOpenAI",
+}
+
+#suppress "@azure-tools/typespec-azure-core/casing-style" "The names of Property types must use PascalCase"
+@added(Versions.v2024_02_15_Preview)
+@doc("Connection details for the LLM resource.")
+model LLMResource {
+  @doc("LLM resource type. The default value is AzureOpenAI.")
+  resourceType?: LLMResourceType = LLMResourceType.AzureOpenAI;
```

Comment:
Please take a look on discriminated type https://typespec.io/docs/standard-library/discriminated-types (it may or may not fit your intention here)

Consider what you would evolve on the API, if later adding a `LLMResourceType.YetAnotherLLM`.

Won't block preview, but need to be finalized before GA.